### ISSUE TITLE
Does not recognize http URLs as links in post

### ISSUE THREAD

**Issue by MightySpaceman** [mastodon/mastodon#22230]
### Pitch

When posting a URL, it will render itself as a clickable hyperlink. However, this only works with https (encrypted) sites. I suggest that this works with http as well

### Motivation

More consistency, and ease of use - so that you don't have to manually copy and paste it into your browser. I doubt it would be that hard to implement either, seeing as there is already the capability to do it with https links

**Comment by rbairwell** [comment]
I'd also like it to autodetect/link ^www\.([a-z09\-\.]{2,})(gtld/tld list)(/([a-z0-9\/\-]*)? style urls (and ask github to add a reminder to not allow idiots to try and type regular expressions on mobile! ;) ) to catch urls missing the schema. (I do know that regex will fail for a few edge cases and it will need to be fixed for uni/punycode urls: but on I'm on mobile)

**Comment by MightySpaceman** [comment]
yes, yes, that would be good as well

It'll probably be adressed someday if this doesn't get through, but the sooner the better

**Comment by eishiya** [comment]
In the web client on my instance, which is running 4.0.2, `http://` URLs are recognised too, and I think always have been. Is this http issue perhaps specific to your client or instance version?

-1 to the regex suggestion. Too many edge cases, I feel, and the TLD list would need to be updated regularly. Plus, sometimes it's useful to mention a URL without linking to it (e.g. when discussing domain names, providing examples, etc), and knowing that it's just a matter of (not) including the protocol is very convenient.

**Comment by ClearlyClaire** [comment]
Do you have an example of an HTTP URL that fails to be recognized? They should definitely work.

**Comment by MightySpaceman** [comment]
https://aus.social/@mightyspaceman/109494524325035068

It might be a matter of it being an IP adress?

**Comment by ClearlyClaire** [comment]
Yeah, this is because it's an IP address, not because it's using HTTP rather than HTTPS.

**Comment by MightySpaceman** [comment]
Ah, I could see how that might fail. Doesn't have a domain suffix like a DNS adress
